lib/ForkProcessSandbox.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "ForkProcessSandbox.h"
+
+#include "TestResult.h"
+
+#include <sys/mman.h>
+#include <unistd.h>
+
+Mutang::ExecutionResult Mutang::ForkProcessSandbox::run(std::function<void (ExecutionResult *)> Func) {
+
+  void *SharedMemory = mmap(NULL,
+                            sizeof(ExecutionResult),
+                            PROT_READ | PROT_WRITE,
+                            MAP_SHARED | MAP_ANONYMOUS,
+                            -1,
+                            0);
+
+  ExecutionResult *SharedResult = new (SharedMemory) ExecutionResult();
+
+  int ChildPid = fork();
+  if (ChildPid == -1) {
+    exit(1);
+  }
+
+  else if (ChildPid == 0) {
+    Func(SharedResult);
+    exit(0);
+  }
+
+  else {
+    /// TODO: Handle at least 3 cases:
+    /// 1) Child process crashes
+    /// 2) Child process exits 1
+    /// 3) Child process hangs
```

Comment:
In mutant I decided to explicitly NOT handle this case from the mutation engine, and decided the test frameworks itself must terminate. The "child process hangs" case is de-facto a halting problem, its inherently unsolvable and only can be mitigated. Timeouts IMO should for that reason than be more close to the source of the intent halting: The test framework.

Also this avoids having to maintain 3 states in reporting, as a "hanged" child process cannot be easily counted as kill or alive.